### PR TITLE
Add Support for Rembrandt platform

### PR DESCRIPTION
**Pull request by reddysujith**
Add SOF driver support on AMD Rembrandt Platform

### CODE REVIEW

File: sound/soc/sof/amd/pci-rmb.c [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+	pdevinfo.id = 0;
+	pdevinfo.parent = &pci->dev;
+	pdevinfo.num_res = ARRAY_SIZE(rembrandt_res);
+	pdevinfo.res = &res[0];
+
+	pdev = platform_device_register_full(&pdevinfo);
+	if (IS_ERR(pdev)) {
+		dev_err(&pci->dev, "cannot register %s device\n", pdevinfo.name);
+		sof_pci_remove(pci);
+		platform_device_unregister(dmic_dev);
```
Comment (plbossart) [inline sound/soc/sof/amd/pci-rmb.c]: does the order matter? This is different from the order in the remove

File: sound/soc/sof/amd/pci-rmb.c [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+};
+
+static void acp_pci_rmb_remove(struct pci_dev *pci)
+{
+	if (dmic_dev)
+		platform_device_unregister(dmic_dev);
+	if (pdev)
+		platform_device_unregister(pdev);
+
+	return sof_pci_remove(pci);
```
Comment (plbossart) [inline sound/soc/sof/amd/pci-rmb.c]: this is a void function, the return is not needed.

File: sound/soc/sof/amd/acp.h [hunk 3]
```diff
@@ -91,9 +91,9 @@ struct dma_descriptor {
 /* Scratch memory structure for communication b/w host and dsp */
 struct  scratch_ipc_conf {
 	/* DSP mailbox */
-	u8 sof_out_box[512];
+	u8 sof_out_box[1024];
 	/* Host mailbox */
-	u8 sof_in_box[512];
+	u8 sof_in_box[1024];
```
Comment (plbossart) [inline sound/soc/sof/amd/acp.h]: this looks like an ABI change, if someone updates their kernel and not the firmware, how would this work?

**Review by plbossart — CHANGES_REQUESTED** [review]

File: sound/soc/sof/amd/pci-rmb.c (hunk 2 above)
Comment (reddysujith) [reply sound/soc/sof/amd/pci-rmb.c]: sof_pci_remove also returns void,,  Anyways i will remove that will be good

File: sound/soc/sof/amd/pci-rmb.c (hunk 1 above)
Comment (reddysujith) [reply sound/soc/sof/amd/pci-rmb.c]: The order doesn't matter as per the code ,But will do to beautify the code

File: sound/soc/sof/amd/acp.h (hunk 3 above)
Comment (reddysujith) [reply sound/soc/sof/amd/acp.h]: Gone through the other part of the code to check some one has written abi code, i see only in core code  the abi versions are written .

In sof firmware they are highlighting ABI label, is that the newer one (which is right side of the label). Is taht can be used here also

any referrence code is appreciated.

File: sound/soc/sof/ipc3-topology.c [hunk 4]
```diff
@@ -1217,6 +1217,36 @@ static int sof_link_acp_sp_load(struct snd_soc_component *scomp, struct snd_sof_
...
 }
 
+static int sof_link_acp_hs_load(struct snd_soc_component *scomp, struct snd_sof_dai_link *slink,
+				struct sof_ipc_dai_config *config, struct snd_sof_dai *dai)
+{
+	struct snd_soc_tplg_hw_config *hw_config = slink->hw_configs;
+	struct sof_dai_private_data *private = dai->private;
+	u32 size = sizeof(*config);
+
+	/* handle master/slave and inverted clocks */
```
Comment (ranj063) [inline sound/soc/sof/ipc3-topology.c]: can you please replace master/slave with inclusive terms?

File: sound/soc/sof/amd/Kconfig [hunk 5]
```diff
@@ -31,4 +31,14 @@ config SND_SOC_SOF_AMD_RENOIR
 	select SND_SOC_SOF_AMD_COMMON
 	help
 	  Select this option for SOF support on AMD Renoir platform
+
```
Comment (ranj063) [inline sound/soc/sof/amd/Kconfig]: what platform driver are you talking about in the commit message?

File: sound/soc/sof/amd/acp-loader.c [hunk 6]
```diff
@@ -198,12 +198,19 @@ EXPORT_SYMBOL_NS(acp_dsp_pre_fw_run, SND_SOC_SOF_AMD_COMMON);
...
 int acp_sof_dsp_run(struct snd_sof_dev *sdev)
 {
+	const struct sof_amd_acp_desc *desc = get_chip_info(sdev->pdata);
 	int val;
 
 	snd_sof_dsp_write(sdev, ACP_DSP_BAR, ACP_DSP0_RUNSTALL, ACP_DSP_RUN);
 	val = snd_sof_dsp_read(sdev, ACP_DSP_BAR, ACP_DSP0_RUNSTALL);
 	dev_dbg(sdev->dev, "ACP_DSP0_RUNSTALL : 0x%0x\n", val);
 
+	/* Some platforms won't support fusion DSP keep offset zero for no support */
```
Comment (ranj063) [inline sound/soc/sof/amd/acp-loader.c]: this sentence is missing a puntuation mark. possibly a period before keep?

File: sound/soc/sof/amd/acp.h (hunk 3 above)
Comment (ranj063) [reply sound/soc/sof/amd/acp.h]: @reddysujith yes, if the corresponsing FW PR for rembrandt support increases the FW ABI, you will need to do the same in the kernel. Also, you'll need to check the FW ABI and keep the original size for older ABI.

File: sound/soc/sof/amd/pci-rmb.c (hunk 1 above)
Comment (reddysujith) [reply sound/soc/sof/amd/pci-rmb.c]: Done

File: sound/soc/sof/amd/pci-rmb.c (hunk 2 above)
Comment (reddysujith) [reply sound/soc/sof/amd/pci-rmb.c]: done

File: sound/soc/sof/amd/acp.h (hunk 3 above)
Comment (reddysujith) [reply sound/soc/sof/amd/acp.h]: In the next patch we are using window info from firmware ,that will solve the problem.if now indow information is available we keep default to 512.

File: sound/soc/sof/ipc3-topology.c (hunk 4 above)
Comment (reddysujith) [reply sound/soc/sof/ipc3-topology.c]: done

File: sound/soc/sof/amd/Kconfig (hunk 5 above)
Comment (reddysujith) [reply sound/soc/sof/amd/Kconfig]: its rembrandt only

File: sound/soc/sof/amd/acp-loader.c (hunk 6 above)
Comment (reddysujith) [reply sound/soc/sof/amd/acp-loader.c]: done

**Comment by plbossart** [comment]
@reddysujith did you intend to close this PR?

**Comment by reddysujith** [comment]
I don't know why..when rebasing and Updating i am getting NEW PR( https://github.com/thesofproject/linux/pull/3749 )

**Pull request closed without merging**